nixos/modules/services/misc/paperless.nix

```diff
@@ -252,9 +252,59 @@ in
         '';
       };
     };
+
+    exporter = {
+
+      enable = lib.mkEnableOption "regular automatic document exports.";
+
+      directory = lib.mkOption {
+        type = lib.types.str;
+        default = cfg.dataDir + "/exports";
+        description = lib.mdDoc "Directory to store exports.";
+      };
+
+      onCalendar = lib.mkOption {
+        type = lib.types.nullOr lib.types.str;
+        default = "01:30:00";
+        description = lib.mdDoc ''
+          When to run the exporter.
+
+          Fills in the OnCalendar section of a timer and uses a systemd.time(7)
+          format.
+
+          `null` disables the timer; allowing you to trigger the
+          `paperless-exporter` service through other means.
+        '';
+      };
+
+      options = lib.mkOption {
+         type = lib.types.attrs;
+         default = {
+          "no-progress-bar" = true;
+          "--no-color" = true;
+          "--compare-checksums" = true;
+          "--delete" = true;
+         };
+         description = lib.mdDoc "Options to pass to the document exporter.";
+      };
+
+      preScript = lib.mkOption {
+         type = lib.types.lines;
+         default = "";
+         description = lib.mdDoc "Script to run right before the export.";
+      };
+
+      postScript = lib.mkOption {
+         type = lib.types.lines;
+         default = "";
+         description = lib.mdDoc "Script to run after finishing the export.";
+      };
```

Comment:
A thought that occured to me since having the discussion is that this technically isn't necessary as you could just make the exporter service dependent on your own custom service that runs before it.

It probably doesn't hurt to have this though.

These should perhaps also document that they're ran inside the dataDir.